### DISCUSSION TITLE
处理生成的window数据集是做什么用的？

### DISCUSSION THREAD

**Discussion by wulongxiaomeem** [pykt-team/pykt-toolkit#48] [Q&A] [answered]
处理生成的window数据集是做什么用的？

**Comment by Liu-lqq** [comment, marked as answer]
window和非window的区别在于学生序列长度大于最大长度x时的处理方式：
    非window直接截断，大于最大长度x的部分会做为全新的学生序列，大于x的部分构成的新序列无法重新利用历史信息。
    window不截断，会使用最近的x-1个历史进行下一时刻的预测。
我们在训练时使用的非window方式，预测时为了充分利用学生历史信息，使用的window方式作为最终预测结果。

**Reply by wulongxiaomeem** [reply]
哇，感谢回复，另外我想问一下我在讨论区发出的另一个问题，为什么训练会出现bug？
